scapy/layers/dns.py

```diff
@@ -19,12 +19,154 @@
 from scapy.sendrecv import sr1
 from scapy.layers.inet import IP, DestIPField, UDP, TCP
 from scapy.layers.inet6 import DestIP6Field
-from scapy.error import warning
+from scapy.error import warning, Scapy_Exception
 from functools import reduce
 import scapy.modules.six as six
 from scapy.modules.six.moves import range
 
 
+def dns_get_str(s, p, pkt=None, _internal=False):
```

Comment:
Could you add some tests that will check the behavior of this function?

Reply:
Could you also document the meaning of _internal ?

Reply:
Added. Also changed a few var names